### PR TITLE
v6.1: update the description for tidb_enable_outer_join_reorder

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

Fix the default value description for `tidb_enable_outer_join_reorder`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/15665
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/12342

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1325,9 +1325,9 @@ This variable is associated with a feature available since v6.2.0 and not usable
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
-- Default value: In v6.1.0, the default value is `ON`. After v6.1.0, the default value is `OFF`.
+- Default value: In v6.1.0, the default value is `ON`.
 - Since v6.1.0, the [Join Reorder](/join-reorder.md) algorithm of TiDB supports Outer Join. This variable controls the support behavior. The default value is `OFF`, which means the Join Reorder's support for Outer Join is disabled by default.
-- For a cluster upgraded from a version earlier than v6.1.0, the default value is `OFF`. For a cluster upgraded from v6.1.0, the default value is `ON`.
+- If the TiDB version before the upgrade is earlier than v6.1.0, the default value of this variable after the upgrade is `ON`.
```
Comment (qiancai) [inline system-variables.md:1330]: The above line reuses the translation in the dev version: https://docs.pingcap.com/tidb/dev/system-variables#tidb_enable_outer_join_reorder-new-in-v610

Comment (qiancai) [inline system-variables.md:1330]:
```suggestion
- If your TiDB cluster is upgraded from an earlier version to v6.1.0,  the default value of this variable after the upgrade is `ON`.
```

File: system-variables.md [hunk 2]
```diff
@@ -1325,9 +1325,9 @@ This variable is associated with a feature available since v6.2.0 and not usable
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
-- Default value: In v6.1.0, the default value is `ON`. After v6.1.0, the default value is `OFF`.
+- Default value: In v6.1.0, the default value is `ON`.
 - Since v6.1.0, the [Join Reorder](/join-reorder.md) algorithm of TiDB supports Outer Join. This variable controls the support behavior. The default value is `OFF`, which means the Join Reorder's support for Outer Join is disabled by default.
-- For a cluster upgraded from a version earlier than v6.1.0, the default value is `OFF`. For a cluster upgraded from v6.1.0, the default value is `ON`.
+- If your TiDB cluster is upgraded from an earlier version to v6.1.0, the default value of this variable after the upgrade is `ON`.
```
Comment (qiancai) [inline system-variables.md:1330]:
```suggestion
- If your TiDB cluster is upgraded from an earlier version to v6.1.0, the default value of this variable after the upgrade is `ON`.  If your TiDB cluster is upgraded from an earlier version to v6.1.1 or a later 6.1 patch version, the default value of this variable after the upgrade is `OFF`.
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**